viewer/packages/tools/src/Timeline/TimelineTool.test.ts

```diff
@@ -0,0 +1,127 @@
+
+    const cadNode = new CadNode(cadMetadata, materialManager);
+    const apiClient = new NodesLocalClient();
+
+    model = new Cognite3DModel(1, 2, cadNode, apiClient);
+
+    timelineTool = new TimelineTool(model);
+
+    kf1Collection = new TreeIndexNodeCollection(new IndexSet([1, 2, 3]));
+    kf1Appearance = { renderGhosted: false };
+    kf2Collection = new TreeIndexNodeCollection(new IndexSet([4, 5, 6]));
+    kf2Appearance = { renderGhosted: true };
+    kf3Collection = new TreeIndexNodeCollection(new IndexSet([7, 8, 9]));
+    kf3Appearance = { renderGhosted: false };
+
+    const kf1 = timelineTool.createKeyframe(new Date('2021-10-25'));
+    const kf2 = timelineTool.createKeyframe(new Date('2021-10-26'));
+    const kf3 = timelineTool.createKeyframe(new Date('2021-10-27'));
+
+    kf1.assignStyledNodeCollection(kf1Collection, kf1Appearance);
+    kf2.assignStyledNodeCollection(kf2Collection, kf2Appearance);
+    kf3.assignStyledNodeCollection(kf3Collection, kf3Appearance);
+  });
+
+  test('play() playback the Timeline', () => {
+    const assignStyledNodeCollectionSpy = jest.spyOn(model, 'assignStyledNodeCollection');
+    const unassignStyledNodeCollectionSpy = jest.spyOn(model, 'unassignStyledNodeCollection');
+    const playSpy = jest.spyOn(timelineTool, 'play');
+
+    timelineTool.play(new Date('2021-10-25'), new Date('2021-10-27'), 30000);
+    const current = TWEEN.now();
+    TWEEN.update(current + 10000);
+
+    expect(assignStyledNodeCollectionSpy).toBeCalledWith(kf1Collection, kf1Appearance);
+    TWEEN.update(current + 20000);
+
+    expect(assignStyledNodeCollectionSpy).toBeCalledWith(kf2Collection, kf2Appearance);
+    TWEEN.update(current + 30000);
+
+    expect(assignStyledNodeCollectionSpy).toBeCalledWith(kf3Collection, kf3Appearance);
+    TWEEN.update();
+
+    expect(unassignStyledNodeCollectionSpy).toBeCalledWith(kf2Collection);
+
+    expect(playSpy).toBeCalledTimes(1);
+  });
+
+  test('stop() stops the Timeline playback', () => {
+    const assignStyledNodeCollectionSpy = jest.spyOn(model, 'assignStyledNodeCollection');
+    const stopSpy = jest.spyOn(timelineTool, 'stop');
+
+    timelineTool.play(new Date('2021-10-25'), new Date('2021-10-27'), 40000);
+    const current = TWEEN.now();
+    TWEEN.update(current + 10000);
+
+    expect(assignStyledNodeCollectionSpy).toBeCalledWith(kf1Collection, kf1Appearance);
+    TWEEN.update(current + 20000);
+
+    expect(assignStyledNodeCollectionSpy).toBeCalledWith(kf2Collection, kf2Appearance);
+    TWEEN.update(current + 30000);
+
+    timelineTool.stop();
+
+    expect(TWEEN.update()).toBeFalse();
+
+    expect(stopSpy).toBeCalledTimes(2);
+  });
+
+  test('pause() & resume() pauses & resumes the Timeline', () => {
+    timelineTool.play(new Date('2021-10-25'), new Date('2021-10-27'), 40000);
+    const pauseSpy = jest.spyOn(timelineTool, 'pause');
+    const resumeSpy = jest.spyOn(timelineTool, 'resume');
```

Comment:
See comment above regarding using spies for stuff you invoke in the test itself.